Sources/Networking/HTTPClient.swift

```diff
@@ -77,9 +77,26 @@ private extension HTTPClient {
 
         var httpRequest: HTTPRequest
         var headers: HTTPClient.RequestHeaders
-        var completionHandler: Completion?
+        var completionHandler: HTTPClient.Completion<Data>?
```

Comment:
This was the key. To be able to store a type erased `Request`, the completion handler just deals with the `Data` itself. The constructor is the only thing that has the `Value` type information, so it handles the deserialization there so everything else in `HTTPClient` is type agnostic.